src/blockchain/alt_block_tree.cpp

```diff
@@ -88,6 +88,26 @@ bool AltTree::acceptBlock(const AltBlock& block, ValidationState& state) {
   return true;
 }
 
+void AltTree::invalidateBlockByHash(const hash_t& blockHash) {
+  index_t* blockIndex = getBlockIndex(blockHash);
+
+  if (blockIndex == nullptr) {
+    // no such block
+    return;
+  }
+
+  invalidateBlockByHash(blockIndex);
+}
+
+void AltTree::invalidateBlockByHash(const index_t* blockIndex) {
+  ValidationState state;
+  bool ret = cmp_.setState(*blockIndex->pprev, state);
+  (void)ret;
+  assert(ret);
+
+  block_index_.erase(blockIndex->getHash());
```

Comment:
how about all blocks after `blockIndex`?

Reply:
Yes, I see it , I will update it.